### PR TITLE
Topic asdplng 68 kingda

### PR DESCRIPTION
**Pull request by v1peractual**
This is adding the ability to use Vault to retieve secrets. This has been tested using the Delta control  on dt-conf01.delta as well as on glick-pup with the standard control repository. This uses the puppet profile secrets to retrieve secrets from Vault.

### CODE REVIEW

File: manifests/kerberos.pp [hunk 1]
```diff
@@ -23,13 +23,24 @@
 # @param root_k5login_principals
 #   Optional Array k5login principals with root privileges
 #
+# @param vaultkeytabkey
+#   # The key to the base64 encoded hostkeytab in vault
+#
+# @param vaultkeytabkey
+#   # The key to the base64 encoded hostkeytab in vault
```
Comment (billglick) [inline manifests/kerberos.pp]: It looks like this docstring is listed twice...

```suggestion
# @param vaultkeytabkey
#   # The key to the base64 encoded hostkeytab in vault
```

File: manifests/kerberos.pp [hunk 2]
```diff
@@ -78,7 +89,19 @@
     }
   }
 
-  if ( $createhostkeytab and $createhostuser )
+  if (lookup(profile_secrets::enable))
+  {
+    $vault_uri = profile_secrets::lookup_uri($vaultsecretdir)
+    $vault_auth = lookup(profile_secrets::vault_authmethod)
+    $vault_kv_version = lookup(profile_secrets::vault_kv_version)
+    $hostkeytabbase64 = Sensitive(vault_key($vault_uri,$vault_auth,$vaultkeytabkey,$vault_kv_version))
```
Comment (billglick) [inline manifests/kerberos.pp]: I think we want to avoid calling this the `hostkeytab`... as that is the keytab file that will later be created. 
Can you continue to use the `createhostkeytab` naming so that `$hostketabbase64` is switched to be `$createhostkeytabbase64`?

And also updated in the few lines below this?

**Review by billglick — CHANGES_REQUESTED** [review]
Besides the 2 suggestions I made to the code...

This is still hard to understand what is going on with Vault. I think it is fine for now, but anything we can do in the following areas would be helpful:
- document more clearly in the README and Wiki documentation
- simplify how the vault secrets are pulled -- I still find it very hard to parse what is going on

While this works for now, we have several other Puppet modules where we want to pull in secrets. Anything we can do to simplify this (or boilerplate this) now would be helpful.

**Comment by v1peractual** [comment]
I understand what you are saying, but I don't think the documentation for secrets needs to be done.  That should be done in https://github.com/ncsa/puppet-profile_secrets. I will add it to my to do list.

**Review by billglick — APPROVED** [review]
Looks good

File: manifests/kerberos.pp (hunk 1 above)
Comment (v1peractual) [reply manifests/kerberos.pp]: This has been fixed

File: manifests/kerberos.pp (hunk 2 above)
Comment (v1peractual) [reply manifests/kerberos.pp]: This has been changed
